routemaster/state_machine/actions.py

```diff
@@ -104,16 +106,16 @@ def process_retries(
         )
 
     for label_name, metadata in relevant_labels.items():
-        label = LabelRef(name=label_name, state_machine=state_machine.name)
-        could_progress = False
+        with process_one_job():
+            label = LabelRef(name=label_name, state_machine=state_machine.name)
+            could_progress = False
 
-        with app.db.begin() as conn:
-            try:
+            with app.db.begin() as conn:
```

Comment:
In fact, I strongly suspect this is an error since `process_one_job` is an instance of `routemaster.cron._process_cron_job._inner` which silences exceptions thrown within the context. This means that if this context errors we'll run `process_transitions(app, label)` for the label anyway. I'm fairly sure we don't want that.

Reply:
We've discussed this IRL and resolved in a refactor I think.